File: app/schemas/admin_trace_lab.py
"""Admin Trace Lab schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field


class AdminTraceRunSummary(BaseModel):
    id: UUID
    conversation_id: UUID
    user_id: UUID
    query: str
    status: str
    parent_run_id: UUID | None = None
    retry_of_run_id: UUID | None = None
    event_count: int = 0
    created_at: datetime
    updated_at: datetime


class AdminTraceRunListResponse(BaseModel):
    items: list[AdminTraceRunSummary]
    total: int
    page: int
    page_size: int


class AdminTraceTimelineItem(BaseModel):
    event_id: str
    event_type: str
    step: int
    sequence_no: int | None = None
    timestamp: str | None = None
    title: str
    kind: str
    status: str
    detail_sanitized: str | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)
    redacted: bool = False


class AdminTraceRunDetailResponse(BaseModel):
    run: AdminTraceRunSummary
    events: list[AdminTraceTimelineItem]
    after_event_id: str | None = None
    anchor_found: bool = True
    last_event_id: str | None = None
    redaction_policy: str

File: front/shared/types/admin-trace-lab.ts
export interface AdminTraceRunSummary {
  id: string
  conversation_id: string
  user_id: string
  query: string
  status: string
  parent_run_id?: string | null
  retry_of_run_id?: string | null
  event_count: number
  created_at: string
  updated_at: string
}

export interface AdminTraceRunListResponse {
  items: AdminTraceRunSummary[]
  total: number
  page: number
  page_size: number
}

export interface AdminTraceTimelineItem {
  event_id: string
  event_type: string
  step: number
  sequence_no?: number | null
  timestamp?: string | null
  title: string
  kind: string
  status: string
  detail_sanitized?: string | null
  metadata: Record<string, unknown>
  redacted: boolean
}

export interface AdminTraceRunDetailResponse {
  run: AdminTraceRunSummary
  events: AdminTraceTimelineItem[]
  after_event_id?: string | null
  anchor_found: boolean
  last_event_id?: string | null
  redaction_policy: string
}
